scripts/iso_mirna_quantification.py

```diff
@@ -0,0 +1,337 @@
+        '--mir-list',
+        help=(
+            "List of miRNA species to have in the different output tables."
+            "Default: %(default)s"
+        ),
+        nargs='*',
+        default=['iso_mirna'],
+        type=str
+    )
+    parser.add_argument(
+        '--lib',
+        help=(
+            "Library to which the alignments belong to. Default : %(deafult)s"
+        ),
+        type=str,
+        default="lib",
+    )
+    parser.add_argument(
+        '-t', '--tag',
+        help=(
+            "Indicate the tag storing the read species name in uppercase."
+            "Default %(default)s."
+        ),
+        default='YW',
+        type=str
+    )
+    parser.add_argument(
+        '-c', '--collapsed',
+        help=(
+            "Indicate that the SAM file has the reads collapsed by sequence"
+            "and alignment. The collapsed name must be build by the alignment"
+            "name followed by a '-' and the number of collpased alignments,"
+            "i.e 1-4. Default %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+    parser.add_argument(
+        '--nh',
+        help=(
+            "Indicate that the SAM file has the NH tag in the read query name."
+            "The name must be build by the alignment name followed by an"
+            "underscore and the NH value, i.e 1-2_4. Default %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+    parser.add_argument(
+        '--len',
+        help=(
+            "Include the species length in the output table."
+            "Default: %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+    parser.add_argument(
+        '--read-ids',
+        help=(
+            "Include read IDs of the alignments that are the same species in"
+            "the output table. Default: %(default)s."
+        ),
+        action='store_true',
+        default=False
+    )
+
+    return parser
+
+
+def get_out_dirs(mir_list: list[str], library: str, outdir: Path) -> Optional[Path]:
+    """Get output directories.
+    
+    Create directories for each miRNA type specified in a list and return a
+    the new directories. If a miRNA type is not found, its corresponding value
+    in the returned list is None.
+
+    Args:
+        mir_list:
+            list with the desired miRNA types tables
+        library:
+            library where the species to be classified belong to
+        outdir:
+            path to the output directory
+    Returns:
+        the new directories and/or None
+    """
+    mir_out = None
+    isomir_out = None
+    iso_mirna_out = None
+
+    if 'mirna' in mir_list:
+        mir_out = outdir/f'mirna_counts_{library}'
+    if 'isomir' in mir_list:
+        isomir_out = outdir/f'isomir_counts_{library}'
+    if 'iso_mirna' in mir_list:
+        iso_mirna_out = outdir/f'iso_mirna_counts_{library}'
+
+    return mir_out, isomir_out, iso_mirna_out
```

Comment:
Why make this so complicated? There's only _one_ output file, that either does or does not contain canonical and/or isomiRs. The user will get what they ask for and see for themselves which feautures are quantified in the output. No need to indicate that in the filename. Just call it `counts_mir_{library}.tsv` )versus `counts_pri-mir_{library}.tsv` for the primary miRNA quantification). When starting your names with the more general term (here: `counts`), in the final workflow output directory, the count tables appear right next to each other.